py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,441 @@
+        The filename of the zcat output file.
+    qn_flag : :class:'bool'
+        Flag if QuasarNP data (or not) was added to the zcat file.
+    sq_flag : :class:'bool'
+        Flag if SQUEzE data (or not) was added to the zcat file.
+    abs_flag : :class:'bool'
+        Flag if MgII Absorption data (or not) was added to the zcat file.
+        
+    Returns
+    -------
+    :class:'str'
+        The filename, with path, of the FITS file written out.
+    """
+    tmark('    Creating file...')
+    full_outputname = os.path.join(outputdir, outputname)
+    
+    prim_hdu = fits.PrimaryHDU()
+    prim_hdu.header['QN_ADDED'] = str(qn_flag)
+    prim_hdu.header['SQ_ADDED'] = str(sq_flag)
+    prim_hdu.header['AB_ADDED'] = str(abs_flag)
+    
+    data_hdu = fits.BinTableHDU.from_columns(zcat,name='ZCATALOG')
+    data_out = fits.HDUList([prim_hdu,data_hdu])
+    
+    data_out.writeto(full_outputname)
+
+    return full_outputname
+
+
+def create_zcat(tile, night, petal_num,
+                zcatdir='/global/cfs/cdirs/desi/spectro/redux/daily', 
+                outputdir='/global/cfs/cdirs/desi/spectro/redux/daily',
+                qn_flag=False, sq_flag=False, abs_flag=False,
+                squeze_model='/global/homes/e/ebie/local/SQUEzE/data/BOSS_train_64plates_model.json'):
+    """This will create a single zcat file from a set of user inputs.
+    
+    Parameters
+    ----------
+    tile : :class:'str'
+        The TILEID of the tile to process.
+    night : :class:'str'
+        The date associated with the observation of the 'tile' used.
+        * Must be in YYYYMMDD format
+    petal_num : :class:'int'
+        If 'all_petals' isn't used, the single petal to create a zcat for.
+    zcatdir : :class:'str'
+        The location for the daily redrock output.
+    outputdir : :class:'str'
+        The filepath to write out the zcat file.
+    qn_flag : :class:'bool'
+        Flag to add QuasarNP data (or not) to the zcat file.
+    sq_flag : :class:'bool'
+        Flag to add SQUEzE data (or not) to the zcat file.
+    abs_flag : :class:'bool'
+        Flag to add MgII Absorption data (or not) to the zcat file.
+    squeze_model : :class:'str' or 'numpy.array'
+        The filename and path for the SQUEzE model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+        
+    Outputs
+    -------
+    A FITS catalog that incorporates redrock, QuasarNP, SQUEzE, and MgII
+    absorption redshifts and confidence values.
+    
+    Notes
+    -----
+    - There will be many here.
+    """
+    # EBL Load the SQUEzE Model file first. This is a very large file,
+    # so if multiple petals are to be processed, we only want to load
+    # it into memory once.
+    if isinstance(squeze_model, str) and sq_flag:
+        tmark('    Loading SQUEzE Model file')
+        squeze_model = Model.from_json(load_json(squeze_model))
```

Comment:
Maybe have this load at the top of the code, so it's only run once at import and is available throughout the module as a global. So, call it `squeze_model` and load it after your initial import statements.

Reply:
It would be better to leave this here, I think. If it loads at import, every time the script is called it will load whether the user wants to use SQUEzE or not. If the user chooses not to use SQUEzE that is a ~6-8 min. time sink for something unused (and a few GB of memory it's carrying around per instance).

Reply:
True, let's leave this load where you have it for now. Depending on whether we definitely choose to use SQUEzE, and how we parelellize, we may want to choose, later on, to load this elsewhere.